scripts/create_deprecated_modules.lean

```diff
@@ -0,0 +1,371 @@
+    match cpts.getLast? with
+    | none => cpts
+    | some last =>
+      cpts.dropLast ++ [if last.endsWith ".lean" then last.dropRight ".lean".length else last]
+  ".".intercalate cpts
+
+#guard mkModName ("Mathlib" / "Data" / "Nat" / "Basic.lean") == "Mathlib.Data.Nat.Basic"
+
+/--
+`deprecateFilePath fname comment` takes as input
+* the path `fname` of a file that was deleted;
+* an optional comment to add in the `deprecated module` syntax.
+
+It returns a pair consisting of
+* an array of `MessageData` giving details about the last time that `fname` was modified
+  and when it was deleted;
+* the content of the deprecated file, matching the original `fname` up to the last import command
+  followed by the `deprecated_module` command with the optional `comment` input, defaulting to
+  `Auto-generated deprecation` if `comment = none`.
+-/
+def deprecateFilePath (fname : String) (rename comment : Option String) :
+    CommandElabM (Array MessageData × String) := do
+  let mut msgs : Array MessageData := #[]
+  -- Check that the input `fname` is a file that currently does not exist.
+  if ← System.FilePath.pathExists fname then
+    throwError m!"The file {fname} exists: I cannot deprecate it!"
+  -- Retrieve the last two commits that modified `fname`:
+  -- the last one is the deletion, the previous one is the last file modification.
+  let log ← runCmd s!"git log --pretty=oneline -2 -- {fname}"
+  let [deleted, lastModified] := log.trim.splitOn "\n" |
+    throwError "Found {(log.trim.splitOn "\n").length} commits, but expected 2! \
+      Please make sure the file {fname} actually exists"
+  let (_deleteHash, deletedMsg) ← processPrettyOneLine deleted "deleted" fname
+  let (modifiedHash, modifiedMsg) ← processPrettyOneLine lastModified "last modified" fname
+  msgs := msgs.push <| m!"The file {fname} was\n"
+  msgs := msgs.push modifiedMsg
+  msgs := msgs.push deletedMsg
+  -- Get the commit date, in `YYYY-MM-DD` format, of the commit deleting the file.
+  let log' ← runCmd s!"git log --format=%cs -2 -- {fname}"
+  let deletionDate := (log'.trim.splitOn "\n")[0]!
+  let deprecation ← mkDeprecationWithDate deletionDate comment
+  msgs := msgs.push ""
+  -- Retrieves the final version of the file, before it was deleted.
+  let file ← runCmd s!"git show {modifiedHash}:{fname}"
+  -- Generate a module deprecation for the file `fname`.
+  let fileHeader := ← match rename with
+    | some rename => do
+      let modName := mkModName rename
+      pure s!"import {modName}\n"
+    | none => getHeader fname file false
+  let deprecatedFile := s!"{fileHeader.trimRight}\n\n{deprecation.pretty.trimRight}\n"
+  msgs := msgs.push <| .trace {cls := `Deprecation} m!"{fname}" #[m!"\n{deprecatedFile}"]
+  return (msgs, deprecatedFile)
+
+elab_rules : command
+| `(#create_deprecated_module%$tk $fnameStx:str $[$comment:str]? $[rename_to $rename?:str]? $[write%$write?]?) => do
+  let fname := fnameStx.getString
+  if ← System.FilePath.pathExists fname then
```

Comment:
This check is duplicated in this rule and the underlying function: remove it here?